gen/solid.hoon

```diff
@@ -11,26 +11,44 @@
   !:
 :-  %say
 |=  $:  {now/@da eny/@uvJ bec/beak}
-        {$~ $~}
+        {arg/$@($~ {top/path $~}) dub/_|}
     ==
 :-  %noun
-=+  top=`path`/(scot %p p.bec)/[q.bec]/(scot %da now)/sys
-=+  pax=`path`(weld top `path`[%hoon ~])
-=+  arp=`path`(weld top `path`[%ovra ~])
+?~  arg  $(arg ~[top=`path`/(scot %p p.bec)/[q.bec]/(scot %da now)/sys])
+=+  pax=`path`(weld top.arg `path`[%hoon ~])
+=+  arp=`path`(weld top.arg `path`[%ovra ~])
 ~&  %solid-start
 =+  txt=.^(@t %cx (weld pax `path`[%hoon ~]))
 =+  rax=.^(@t %cx (weld arp `path`[%hoon ~]))
 =+  ^=  ken
-    =-  ?:(?=($& -.res) p.res ((slog p.res) ~))
+    =-  ?:(?=($& -.res) p.res (mean (flop p.res)))
     ^=  res  %-  mule  |.
     ~&  %solid-loaded
     =+  gen=(rain pax txt)
     ~&  %solid-parsed
     =+  one=(~(mint ut %noun) %noun gen)
     ~&  %solid-compiled
-    =+  two=(~(mint ut p.one) %noun (rain arp rax))
-    ~&  %solid-arvo
-    [7 q.one q.two]
+    ?.  dub
+      =+  two=(~(mint ut p.one) %noun (rain arp rax))
+      ~&  %solid-arvo
+      [7 q.one q.two]
```

Comment:
I think variable names indicating that `one` is really `hoon` and `two` is really `arvo` would be helpful.